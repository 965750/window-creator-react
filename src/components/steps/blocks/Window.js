import React, { Component } from 'react'
import styles from './Window.module.scss'

class Window extends Component {
    state = {
        rows: ['','',''],
        columns: ['','','']
    }

    componentDidMount () {
        setTimeout(() => {
            this.setState({
                rows: Array.apply(null, Array(5))
            })
        }, 3000)
    }

    render () {
        const table = this.state.rows.map((row, index) => {
            return (
                <tr className={`${styles.window__rows}`} key={index}>
                    { this.state.columns.map((column, index) => {
                        return (
                            <td className={`${styles.window__columns}`} key={index}>&nbsp;</td>
                        )
                    }) }
                </tr>
            )
        })

        return (
            <div className='h-full z-20' style={{ width: `${this.props.width}px`, height: `${this.props.height}px` }}>
                <table className={`${styles.window} w-full h-full`}>
                    <tbody>
                        { table }
                    </tbody>
                </table>
            </div>
        )
    }
}

export default Window
