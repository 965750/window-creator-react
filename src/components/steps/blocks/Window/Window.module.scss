.window {
    border-collapse: collapse;
  }
  
.window, .window__rows, .window__columns {
    border: 3px solid;
}