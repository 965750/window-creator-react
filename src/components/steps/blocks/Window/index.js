import PropTypes from 'prop-types'
import React from 'react'
import styles from './Window.module.scss'

const Window = ({ window, innerWidth }) => {
  const table = window.rows.map((row, index) => (
    <tr
      style={{ borderColor: window.color }}
      className={`${styles.window__rows}`}
      key={index} // eslint-disable-line
    >
      {window.columns.map((
        column,
        index // eslint-disable-line
      ) => (
        <td
          style={{
            borderColor: window.color,
          }}
          className={`${styles.window__columns}`}
          key={index} // eslint-disable-line
        >
          &nbsp;
        </td>
      ))}
    </tr>
  ))

  return (
    <div
      className="h-full z-20"
      style={{
        width: `${window.width * (innerWidth > 767 ? 1 : 0.6)}px`,
        height: `${window.height * (innerWidth > 767 ? 1 : 0.6)}px`,
      }}
    >
      <table
        style={{ borderColor: window.color }}
        className={`${styles.window} w-full h-full`}
      >
        <tbody>{table}</tbody>
      </table>
    </div>
  )
}

Window.propTypes = {
  innerWidth: PropTypes.number,
  window: PropTypes.shape({
    innerWidth: PropTypes.number,
    color: PropTypes.string,
    width: PropTypes.number,
    height: PropTypes.number,
    doorType: PropTypes.number,
    rows: PropTypes.array,
    columns: PropTypes.array,
  }),
}

export default Window
