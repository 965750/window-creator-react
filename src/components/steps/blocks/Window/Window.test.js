import '@testing-library/jest-dom/extend-expect'

import { applyMiddleware, compose, createStore } from 'redux';
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { render, cleanup } from '@testing-library/react';

import Window from './index'
import messages from '../../../../messages-i18n'
import thunk from 'redux-thunk'
import React from 'react'

const initialStore = {
  theme: {
    lang: 'pl',
  },
}

const reducer = (state = initialStore, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const store = createStore(
  reducer,
  compose(
    applyMiddleware(thunk),
  ),
)

const renderWithRedux = (component) => ({
  ...render(<Provider store={store}>{component}</Provider>),
})

describe('Window', () => {
  let wrapper
  const initialProps = {
    window: {
      color: '#000',
      width: 90,
      height: 180,
      doorType: 2,
      rows: [0, 1],
      columns: [0, 1],
    },
  }

  const setUp = (props = initialProps) => {
    wrapper = renderWithRedux(
      <IntlProvider
        locale="en"
        messages={messages[store.getState().theme.lang]}
      >
        <Window {...props} />
      </IntlProvider>,
    )
  }

  afterEach(cleanup)

  it('should match snapshot', () => {
    setUp()

    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
