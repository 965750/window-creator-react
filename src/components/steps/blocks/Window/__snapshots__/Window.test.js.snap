// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Window should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-full z-20"
    style="width: 54px; height: 108px;"
  >
    <table
      class="window w-full h-full"
      style="border-color: #000;"
    >
      <tbody>
        <tr
          class="window__rows"
          style="border-color: #000;"
        >
          <td
            class="window__columns"
            style="border-color: #000;"
          >
             
          </td>
          <td
            class="window__columns"
            style="border-color: #000;"
          >
             
          </td>
        </tr>
        <tr
          class="window__rows"
          style="border-color: #000;"
        >
          <td
            class="window__columns"
            style="border-color: #000;"
          >
             
          </td>
          <td
            class="window__columns"
            style="border-color: #000;"
          >
             
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
